dali/pipeline/pipeline.h

```diff
@@ -162,27 +162,10 @@ class DLL_PUBLIC Pipeline {
                               std::optional<std::string> data_id, AccessOrder order = {},
                               InputOperatorSettingMode ext_src_setting_mode = {},
                               bool is_refeeding = false) {
-    OpType op_type;
-    OpNodeId node_id = -1;
-
-    if (graph_.TensorExists(name + "_cpu")) {
-      node_id = graph_.TensorSourceID(name + "_cpu");
-      op_type = graph_.NodeType(node_id);
-      DALI_ENFORCE(op_type == OpType::CPU,
-                   "Internal error setting external input data.");
-    } else if (graph_.TensorExists(name + "_gpu")) {
-      node_id = graph_.TensorSourceID(name + "_gpu");
-      op_type = graph_.NodeType(node_id);
-      DALI_ENFORCE(op_type == OpType::GPU || op_type == OpType::MIXED,
-                   "Internal error setting external input data.");
-    } else {
-      DALI_FAIL("Cannot find " + name + " tensor, it doesn't exists or was pruned as unused one.");
-    }
-
-    auto &node = graph_.Node(node_id);
-    OperatorBase *op_ptr = &node.InstantiateOperator();
+    auto *node = GetOperatorNode(name);
```

Comment:
I think it linearly traverses all nodes, maybe we can collect all InputOps in `DiscoverInputOperators` and traverse over it?
Just a thought (with AutoAug we may have a lot of nodes).